// @connectrpc/connect-query/dist/cjs/index.js
"use strict";
// Copyright 2021-2023 The Connect Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.useMutation = exports.useSuspenseQuery = exports.useQuery = exports.useSuspenseInfiniteQuery = exports.useInfiniteQuery = exports.TransportProvider = exports.useTransport = void 0;
__exportStar(require("@connectrpc/connect-query-core"), exports);
var use_transport_js_1 = require("./use-transport.js");
Object.defineProperty(exports, "useTransport", { enumerable: true, get: function () { return use_transport_js_1.useTransport; } });
Object.defineProperty(exports, "TransportProvider", { enumerable: true, get: function () { return use_transport_js_1.TransportProvider; } });
var use_infinite_query_js_1 = require("./use-infinite-query.js");
Object.defineProperty(exports, "useInfiniteQuery", { enumerable: true, get: function () { return use_infinite_query_js_1.useInfiniteQuery; } });
Object.defineProperty(exports, "useSuspenseInfiniteQuery", { enumerable: true, get: function () { return use_infinite_query_js_1.useSuspenseInfiniteQuery; } });
var use_query_js_1 = require("./use-query.js");
Object.defineProperty(exports, "useQuery", { enumerable: true, get: function () { return use_query_js_1.useQuery; } });
Object.defineProperty(exports, "useSuspenseQuery", { enumerable: true, get: function () { return use_query_js_1.useSuspenseQuery; } });
var use_mutation_js_1 = require("./use-mutation.js");
Object.defineProperty(exports, "useMutation", { enumerable: true, get: function () { return use_mutation_js_1.useMutation; } });
