// @connectrpc/connect-query-core/dist/cjs/index.js
"use strict";
// Copyright 2021-2023 The Connect Authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
Object.defineProperty(exports, "__esModule", { value: true });
exports.skipToken = exports.addStaticKeyToTransport = exports.createQueryOptions = exports.createInfiniteQueryOptions = exports.callUnaryMethod = exports.createProtobufSafeUpdater = exports.createConnectQueryKey = void 0;
var connect_query_key_js_1 = require("./connect-query-key.js");
Object.defineProperty(exports, "createConnectQueryKey", { enumerable: true, get: function () { return connect_query_key_js_1.createConnectQueryKey; } });
var utils_js_1 = require("./utils.js");
Object.defineProperty(exports, "createProtobufSafeUpdater", { enumerable: true, get: function () { return utils_js_1.createProtobufSafeUpdater; } });
var call_unary_method_js_1 = require("./call-unary-method.js");
Object.defineProperty(exports, "callUnaryMethod", { enumerable: true, get: function () { return call_unary_method_js_1.callUnaryMethod; } });
var create_infinite_query_options_js_1 = require("./create-infinite-query-options.js");
Object.defineProperty(exports, "createInfiniteQueryOptions", { enumerable: true, get: function () { return create_infinite_query_options_js_1.createInfiniteQueryOptions; } });
var create_query_options_js_1 = require("./create-query-options.js");
Object.defineProperty(exports, "createQueryOptions", { enumerable: true, get: function () { return create_query_options_js_1.createQueryOptions; } });
var transport_key_js_1 = require("./transport-key.js");
Object.defineProperty(exports, "addStaticKeyToTransport", { enumerable: true, get: function () { return transport_key_js_1.addStaticKeyToTransport; } });
var query_core_1 = require("@tanstack/query-core");
Object.defineProperty(exports, "skipToken", { enumerable: true, get: function () { return query_core_1.skipToken; } });
